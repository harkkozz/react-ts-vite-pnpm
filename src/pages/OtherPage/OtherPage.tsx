import React from 'react';

import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router';

import Button from 'components/Button/Button';

import styles from './OtherPage.module.scss';

const OtherPage: React.FC<React.PropsWithChildren> = () => {
  const { t } = useTranslation('');

  const navigate = useNavigate();

  return (
    <div>
      <h2 className={styles.heading}>{t('otherPage')}</h2>
      <Button text={t('homePage')} onClick={() => navigate('/')} />
    </div>
  );
};

export default OtherPage;
