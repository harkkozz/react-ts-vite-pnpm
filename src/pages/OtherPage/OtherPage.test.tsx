import { fireEvent, render } from '@testing-library/react';
import Test from 'testUtils/TestApp';
import { describe, expect, it, vi } from 'vitest';

import OtherPage from './OtherPage';

const onUseNavigate = vi.fn();

describe('Other page', () => {
  it('Should match snapshot', () => {
    const { baseElement } = render(
      <Test path="/other-page">
        <OtherPage />
      </Test>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Should navigate to home page', () => {
    const { getAllByRole } = render(
      <Test path="/other-page">
        <OtherPage />
      </Test>
    );
    const onNextPage = getAllByRole('button', { name: 'Home page' });
    fireEvent.click(onNextPage[0]);
    expect(onUseNavigate).toHaveBeenCalledTimes(0);
  });
});
