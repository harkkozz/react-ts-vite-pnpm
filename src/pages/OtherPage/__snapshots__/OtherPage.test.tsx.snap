// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Other page > Should match snapshot 1`] = `
<body>
  <div>
    <div>
      <h2
        class="_heading_2aa34a"
      >
        Other page
      </h2>
      <button
        class="_link_2aa34a"
      >
        Home page
      </button>
    </div>
  </div>
</body>
`;
